src/Position/Position.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Position;
+
+/**
+ * Provides a standard reusable implementation of `Positionable`.
+ *
+ * @internal
+ *
+ * @phpstan-require-implements Positionable
+ */
+trait Position
+{
+    /**
+     * @var int<1, max>|null
+     */
+    protected $lineNumber;
+
+    /**
+     * @var int<0, max>|null
+     */
+    protected $columnNumber;
+
+    /**
+     * @return int<1, max>|null
+     */
+    public function getLineNumber(): ?int
+    {
+        return $this->lineNumber;
+    }
+
+    /**
+     * @return int<0, max>
+     */
+    public function getLineNo(): int
+    {
+        return $this->getLineNumber() ?? 0;
+    }
+
+    /**
+     * @return int<0, max>|null
+     */
+    public function getColumnNumber(): ?int
+    {
+        return $this->columnNumber;
+    }
+
+    /**
+     * @return int<0, max>
+     */
+    public function getColNo(): int
+    {
+        return $this->getColumnNumber() ?? 0;
+    }
+
+    /**
+     * @param int<0, max>|null $lineNumber
```

Comment:
I propose we directly make this non-zero (also in the interface). Or am I missing something here?

```suggestion
     * @param int<1, max>|null $lineNumber
```

Reply:
Passing `0` is deprecated, but we need to allow it until it is no longer supported.  The DocBlock comment for this parameter of the method in `Positionable` has a deprecation notice to that effect.  I didn't include a deprecation notice in the trait, because it's internal.

Reply:
Ah, I had missed that `setPosition` already gets called. Then scratch my comments. :-)

Reply:
And it's part of the public API.